### ISSUE TITLE
NTPClient::begin() doesn't check that the UDP object successfully initialized.

### ISSUE THREAD

**Issue by MHotchin** [arduino-libraries/NTPClient#83]
The underlying _udp object can fail its own 'begin()' call if no sockets are available.

Have hit this case on Wiznet 5100 hardware with a webserver also running.

**Comment by workpage2** [comment]
How did you solve the problem?

**Comment by WhymustIhaveaname** [comment]
The begin function of NTPC look like
```
void NTPClient::begin(int port) {
  this->_port = port;

  this->_udp->begin(this->_port);

  this->_udpSetup = true;
}
```
So you can check UDP faterewards manually.

**Comment by MHotchin** [comment]
Code change as follows:

void NTPClient::begin(int port) {
  this->_port = port;

  this->_udpSetup = this->_udp->begin(this->_port);
}

**Comment by workpage2** [comment]
Where to change this code?

**Comment by WhymustIhaveaname** [comment]
Download .h and .cpp file of this repo in to your project folder. Change it as MHotchin said. Change the include sentence of your .ino from 
```
#include<NTPClient.h>
```
to
```
#include"NTPClient.h"
```

**Comment by WhymustIhaveaname** [comment]
But remember, If you use MHotchin's method and if the UDP initialize fails, the `this->_udpSetup` flag will be set to `false`. Refering to `update` of NTPC
```
bool NTPClient::update() {
  if ((millis() - this->_lastUpdate >= this->_updateInterval)     // Update after _updateInterval
    || this->_lastUpdate == 0) {                                // Update if there was no update yet.
    if (!this->_udpSetup) this->begin();                         // setup the UDP client if needed
    return this->forceUpdate();
  }
  return false;   // return false if update does not occur
}
```
, we know that udp will only be intialized again. So I recommand you to use a while loop to make sure UDP is indeed inited.

**Comment by MHotchin** [comment]
No sense using a while() loop, if ‘begin()’ fails you should wait some time *then* try again.

Your code should always handle the case that ‘update()’ fails, so if ‘begin()’ fails just do nothing – let ‘update()’ try to fix it, and if that fails then wait until your next ‘update()’ time and try again.

Quite frankly, the error handling and design of the code in NTP needs *serious* work.  For example, this whole problem would *go away* if the UDP client was ephemeral instead of bound to the lifetime of the UDP client.  This would also free up a socket for use for the 99.9% of the time that NTP wasn’t actually doing an update.   I’d do it, but without some consensus that these changes would be integrated I’m not going to cmmit my time.

**Comment by WhymustIhaveaname** [comment]
Yes, of course he should delay a short time in the while loop if failed. What's more, for his case, he can do something that makes initialize UDP more easily, such as free some port.
